### DISCUSSION TITLE
How to access env in nuxt config

### DISCUSSION THREAD

**Discussion by freddy-daniel** [nuxt/nuxt#20517] [Questions]
I know we can access env variables through runtimeConfig in plugins, middlware ...etc. But how to get env variables in same nuxt.config.ts. I'm trying to read process.env.NUXT_PUBLIC_API_BASE_URL_PROXY in same file but it's treated as falsy value

Nuxt version: 
  "nuxt": "^3.4.2"

.env
```
NUXT_PUBLIC_API_PATH=/api/v1
NUXT_PUBLIC_API_BASE_URL_PROXY=https://example.com
```

nuxt.config.ts
``` 
export default defineNuxtConfig({
  ssr: false,
  runtimeConfig: {
    public: {
      apiPath: '',
      apiBaseUrlProxy: '',
      identityUrl: ''
    }
  },
  nitro: {
    devProxy: {
      '/api/v1': {
        target:
          process.env.NUXT_PUBLIC_API_BASE_URL_PROXY
            ? `${process.env.NUXT_PUBLIC_API_BASE_URL_PROXY}${process.env.NUXT_PUBLIC_API_PATH}`
            : 'https://example2.com/api/v1',
        changeOrigin: true,
        prependPath: true,
        secure: false
      }
    }
  }
});
```

**Comment by AloisSeckar** [comment]
I was dealing with quite similar question recently (configuration of Redis storage in Nitro) and the answer was: As of now it can't be done directly inside `nuxt.config.ts`, but you can define Nitro plugin and access `useRuntimeConfig()` from there.

Explanation and example here:
https://github.com/unjs/nitro/issues/1161#issuecomment-1511444675
